internal/app/machined/internal/phase/config/config.go

```diff
@@ -32,5 +35,20 @@ func (task *Task) standard(r runtime.Runtime) (err error) {
 		return err
 	}
 
+	// Detect if config is a gzip archive and unzip it if so
+	contentType := http.DetectContentType(b)
+	if contentType == "application/x-gzip" {
+		gzipReader, err := gzip.NewReader(bytes.NewBuffer(b))
```

Comment:
`bytes.NewReader` might be more appropriate here (instead of `bytes.NewBuffer`) as it's a bit more lightweight